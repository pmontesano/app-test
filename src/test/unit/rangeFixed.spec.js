/**
 * @jest-environment jsdom
 */

import React from 'react';
import { create } from 'react-test-renderer';
import { mount, shallow, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import RangeFixed from '../../components/rangeFixed';

configure({ adapter: new Adapter() });

const props = {
  min: 1.99,
  max: 70.99,
  values: [1.99, 5.99, 10.99, 30.99, 50.99, 70.99],
  onChange: jest.fn(),
};

describe('RangeFixed', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<RangeFixed {...props} />);
  });

  it('should render correctly', () => {
    const wrapper = create(<RangeFixed {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('initial render values min and max must be equal 1.99 and 70.99', () => {
    const wrapper = mount(<RangeFixed {...props} />);

    expect(wrapper.props().min).toEqual(1.99);
    expect(wrapper.props().max).toEqual(70.99);

    expect(wrapper.props().values).toEqual([
      1.99, 5.99, 10.99, 30.99, 50.99, 70.99,
    ]);
  });

  it('should render label', () => {
    const wrapper = shallow(<RangeFixed {...props} />);
    expect(wrapper.find('.range-slider-label')).toHaveLength(2);
  });

  it('should render buttons', () => {
    const wrapper = shallow(<RangeFixed {...props} />);
    expect(wrapper.find('button')).toHaveLength(2);
  });

  it('should render track', () => {
    const wrapper = shallow(<RangeFixed {...props} />);
    expect(wrapper.find('.range-slider-fixed__track')).toHaveLength(1);
  });
});
