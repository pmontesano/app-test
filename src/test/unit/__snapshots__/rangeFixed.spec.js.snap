// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeFixed should render correctly 1`] = `
<div
  className="range-slider-wrapper range-slider-fixed"
>
  <div
    className="range-slider-label range-slider-label--left"
  >
    1.99
     €
  </div>
  <div
    className="range-slider"
  >
    <div
      className="range-slider__thumb"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      <button
        className="range-slider__handle range-slider__handle--left range-slider__handle-fixed--left "
        style={
          Object {
            "zIndex": false,
          }
        }
        value={1}
      />
    </div>
    <div
      className="range-slider__thumb"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      <button
        className="range-slider__handle range-slider__handle--right  range-slider__handle-fixed--right "
        style={
          Object {
            "left": "83%",
          }
        }
        value={100}
      />
    </div>
    <div
      className="range-slider-fixed__track"
    >
      <svg
        height="10"
        role="presentation"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          className="range-slider-fixed__tick"
          height="10"
          value={1.99}
          width="1"
          x="0%"
          y="3"
        />
        <rect
          className="range-slider-fixed__tick"
          height="10"
          value={5.99}
          width="1"
          x="16%"
          y="3"
        />
        <rect
          className="range-slider-fixed__tick"
          height="10"
          value={10.99}
          width="1"
          x="33%"
          y="3"
        />
        <rect
          className="range-slider-fixed__tick"
          height="10"
          value={30.99}
          width="1"
          x="50%"
          y="3"
        />
        <rect
          className="range-slider-fixed__tick"
          height="10"
          value={50.99}
          width="1"
          x="66%"
          y="3"
        />
        <rect
          className="range-slider-fixed__tick"
          height="10"
          value={70.99}
          width="1"
          x="83%"
          y="3"
        />
      </svg>
    </div>
  </div>
  <div
    className="range-slider-label range-slider-label--right"
  >
    70.99
     €
  </div>
</div>
`;
